rest-api/src/main/java/gov/cms/qpp/conversion/api/services/CpcFileServiceImpl.java

```diff
@@ -45,13 +46,30 @@
 	 */
 	public InputStreamResource getFileById(String fileId) throws IOException {
 		Metadata metadata = dbService.getMetadataById(fileId);
-		if (metadata != null && metadata.getCpc() && !metadata.getCpcProcessed()) {
+		if (isAnUnprocessedCpcFile(metadata)) {
 			return new InputStreamResource(storageService.getFileByLocationId(metadata.getSubmissionLocator()));
 		} else {
 			throw new NoFileInDatabaseException(FILE_NOT_FOUND);
 		}
 	}
 
+	/**
+	 * Process to ensure the file is an unprocessed cpc+ file and marks the file as processed
+	 *
+	 * @param fileId Identifier of the CPC+ file
+	 * @return Success or failure message.
+	 */
+	public String processFileById(String fileId) {
+		Metadata metadata = dbService.getMetadataById(fileId);
+		if (isAnUnprocessedCpcFile(metadata)) {
+			metadata.setCpcProcessed(true);
+			dbService.write(metadata);
```

Comment:
This returns a `CompletableFuture<Metadata>`, so please use that variable and call `.join()` on it if you want it to complete before returning.  Unless we don't care if it completes the DynamoDB write before returning `FILE_FOUND` as the response?

Reply:
No, you're correct and I overlooked this. Sad day for sam

Reply:
Unfortunately, I don't know what exactly I'll use the variable the write returns for. I don't really need the metadata after it's been updated it in the database.